### PR TITLE
Fix recent FuzzBench cloud experiment failures

### PR DESCRIPTION
**Pull request by DonggeLiu**
1. Fix `TypeError: expected str, bytes or os.PathLike object, not NoneType` in [`2024-08-10-test`](https://github.com/google/fuzzbench/pull/2020#issuecomment-2280854078).
```python
Traceback (most recent call last):
  File "/src/experiment/runner.py", line 468, in experiment_main
    runner.conduct_trial()
  File "/src/experiment/runner.py", line 290, in conduct_trial
    self.set_up_corpus_directories()
  File "/src/experiment/runner.py", line 275, in set_up_corpus_directories
    _unpack_clusterfuzz_seed_corpus(target_binary, input_corpus)
  File "/src/experiment/runner.py", line 144, in _unpack_clusterfuzz_seed_corpus
    seed_corpus_archive_path = get_clusterfuzz_seed_corpus_path(
  File "/src/experiment/runner.py", line 98, in get_clusterfuzz_seed_corpus_path
    fuzz_target_without_extension = os.path.splitext(fuzz_target_path)[0]
  File "/usr/local/lib/python3.10/posixpath.py", line 118, in splitext
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```
This happens on [many benchmarks+fuzzers](https://pantheon.corp.google.com/logs/query;query=%222024-08-10-test%22%0Aseverity%3E%3DERROR%0A--Hide%20similar%20entries%0A-%2528jsonPayload.message%3D~%22Error%20watching%20metadata:%20context%20canceled%22%2529%0A--End%20of%20hide%20similar%20entries;cursorTimestamp=2024-08-10T11:04:34.735815901Z;duration=P7D?project=fuzzbench&mods=logs_tg_prod).
To be investigated later:
1. Why `fuzz_target_path` is `None`.
2. Why this did not happen in other recent experiments.
3. I thought I had seen this a long ago, Déjà vu?

2. Fixing `No such file or directory: '/work/measurement-folders/<benchmark>-<fuzzer>/merged.json`:
```python
Traceback (most recent call last):
  File "/work/src/experiment/measurer/coverage_utils.py", line 74, in generate_coverage_report
    coverage_reporter.generate_coverage_summary_json()
  File "/work/src/experiment/measurer/coverage_utils.py", line 141, in generate_coverage_summary_json
    result = generate_json_summary(coverage_binary,
  File "/work/src/experiment/measurer/coverage_utils.py", line 269, in generate_json_summary
    with open(output_file, 'w', encoding='utf-8') as dst_file:
FileNotFoundError: [Errno 2] No such file or directory: '/work/measurement-folders/lcms_cms_transform_fuzzer-centipede/merged.json'
```

3. Remove incompatible benchmarks: `openh264_decoder_fuzzer`, `stb_stbi_read_fuzzer`

### CODE REVIEW

**Comment by DonggeLiu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2024-08-10-test --fuzzers aflplusplus centipede honggfuzz libfuzzer --benchmarks stb_stbi_read_fuzzer openh264_decoder_fuzzer

**Comment by DonggeLiu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2024-08-12-dg --fuzzers aflplusplus centipede honggfuzz libfuzzer --benchmarks stb_stbi_read_fuzzer openh264_decoder_fuzzer

**Comment by DonggeLiu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2024-08-12-2023 --fuzzers aflplusplus centipede honggfuzz libfuzzer --benchmarks stb_stbi_read_fuzzer openh264_decoder_fuzzer

**Comment by DonggeLiu** [comment]
Experiment `2024-08-12-2023` data and results will be available later at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2024-08-12-2023/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/2024-08-12-2023/index.html).
[The experiment report(experimental)](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2024-08-12-2023/index.html).

**Comment by DonggeLiu** [comment]
This failed likely because both fuzz targets failed to build with all fuzzers, e.g.:
image

**Comment by DonggeLiu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2024-08-13-2023-libfuzzer --fuzzers libfuzzer

**Comment by DonggeLiu** [comment]
/gcbrun run_experiment.py -a --experiment-config /opt/fuzzbench/service/experiment-config.yaml --experiment-name 2024-08-13-2023-libfuzzer-1 --fuzzers libfuzzer

**Comment by DonggeLiu** [comment]
Experiment `2024-08-13-2023-libfuzzer-1` data and results will be available later at:
[The experiment data](https://storage.googleapis.com/fuzzbench-data/index.html?prefix=2024-08-13-2023-libfuzzer-1/).
[The experiment report](https://storage.googleapis.com/www.fuzzbench.com/reports/2024-08-13-2023-libfuzzer-1/index.html).
[The experiment report(experimental)](https://storage.googleapis.com/www.fuzzbench.com/reports/experimental/2024-08-13-2023-libfuzzer-1/index.html).

**Comment by DonggeLiu** [comment]
Report is back : ) @addisoncrump
I will wait a bit longer before merging this to ensure the report stays alive.
Once I merge this to master, could you please update your PR and bring back the changes you added?
Thanks!

**Comment by addisoncrump** [comment]
Sure, I'll rebase.

**Comment by addisoncrump** [comment]
@DonggeLiu I am able to build both openh264 and stb_stbi fuzzers as in `master` locally with no issue. Like #2021, I think this is a cache issue.

**Comment by DonggeLiu** [comment]
> @DonggeLiu I am able to build both openh264 and stb_stbi fuzzers as in `master` locally with no issue. Like #2021, I think this is a cache issue.

I see, thanks for the info!
Given that you are [investigating this](https://github.com/google/fuzzbench/pull/2021#issuecomment-2283966848), is there any help I can provide?
For example, if you think some more cloud build logs can save you time debugging, please feel free to add them and request experiments.
I can run them for you and send you the related logs.

**Comment by DonggeLiu** [comment]
Report on this PR is still not ready, likely due to some VMs were preemptied.
I will give it one more day just to be 100% safe.

**Comment by addisoncrump** [comment]
> Given that you are investigating this, is there any help I can provide?

Ah, I was investigating the specific issue with the bug benchmark. I don't think I can offer much help with the CI or the fuzzbench infra directly. I can say, however, that the coverage benchmarks you removed *do* work as expected locally with test-run. I need to check if the coverage measurer works as anticipated; maybe this needs to be updated instead.

**Comment by addisoncrump** [comment]
Ah, @DonggeLiu, try running `make test-run-coverage-all`. It complains that it can't find `bloaty_fuzz_target` :eyes:

**Comment by tokatoka** [comment]
> @DonggeLiu I am able to build both openh264 and stb_stbi fuzzers as in master locally with no issue. Like https://github.com/google/fuzzbench/pull/2021, I think this is a cache issue.

For me the same, they are working. I don't think they should be removed

**Comment by DonggeLiu** [comment]
I see, thanks @addisoncrump and @tokatoka .
I've brought them back.

The experiment is about to finish, I will merge this tmr morning.

**Comment by addisoncrump** [comment]
I confirmed the coverage measurers build locally as well. Will test when everything has finished building.

**Comment by addisoncrump** [comment]
Yup, I tested openh264 and stb benchmarks locally and they do perform measurements as anticipated. The issue is with the GCP runs, I would presume a build cache issue.

**Comment by DonggeLiu** [comment]
> Yup, I tested openh264 and stb benchmarks locally and they do perform measurements as anticipated. The issue is with the GCP runs, I would presume a build cache issue.

I see, I reckon this could be due to impatible GCP vm environment and llvm?
I will look into this once I finish other tasks in hand.

**Comment by DonggeLiu** [comment]
Thanks for the information again, @addisoncrump!

**Comment by DonggeLiu** [comment]
TBR by @jonathanmetzman.

The experiment that proving this works:
https://github.com/google/fuzzbench/pull/2023#issuecomment-2285147301

**Comment by addisoncrump** [comment]
>  When you test them locally, did you remove their old local images beforehand?

Yes, I do a `docker system prune --all` before every experiment.

**Comment by DonggeLiu** [comment]
> > When you test them locally, did you remove their old local images beforehand?
> 
> Yes, I do a `docker system prune --all` before every experiment.

I see, thanks for confirming.
I will merge this then.

File: experiment/runner.py [hunk 1]
```diff
@@ -95,6 +95,8 @@ def _clean_seed_corpus(seed_corpus_dir):
 def get_clusterfuzz_seed_corpus_path(fuzz_target_path):
     """Returns the path of the clusterfuzz seed corpus archive if one exists.
     Otherwise returns None."""
+    if not fuzz_target_path:
+        return None
```
Comment (DonggeLiu) [inline experiment/runner.py:99]: Add an error log here because this is unexpected.

Comment (tokatoka) [reply experiment/runner.py:99]: One question, why is this function even called?

https://github.com/google/fuzzbench/blob/master/experiment/runner.py#L277
I think this is the line that eventually calls this line. But for example, when we observed the error for addison's experiment, the ossfuzz corpus was *NOT* used right? (unless they specified `oss-fuzz-corpus: true`)
then why we would unpack the clusterfuzz seed corpus at all?

**Comment by addisoncrump** [comment]
Just to reiterate, this is a major threat to validity -- especially when cached data is used. [The cache completely overwrites the report](https://www.fuzzbench.com/reports/2024-08-25-2039-bases-1), so the final report generated is simply showing only the last successful experiment. This effectively invalidates all future Fuzzbench reports until this issue is resolved.

I think the report generation issue indicates that safeguards should be put in place that simply terminate the experiment in such degenerative cases, since the results are effectively guaranteed to be invalid.

**Pull request merged**